cirq/api/google/v1/hardware.proto

```diff
@@ -0,0 +1,92 @@
+import "cirq/api/google/v1/params.proto";
+
+
+package cirq.api.google.v1;
+
+option java_package = "com.google.cirq.api.google.v1";
+option java_outer_classname = "HardwareProto";
+option java_multiple_files = true;
+
+// The operations supported by the hardware and the current performance
+// characteristics of these operations.
+message HardwareSpec {
+  // The version name of the hardware.
+  //
+  // Currently supports: `bristlecone`.
+  string processor_name = 1;
+
+  // The supported operations on the hardware. This information is static
+  // and does not change for a given processor_name.
+  GateSet gate_set = 2;
+
+  // The performance characteristics of the hardware, recorded at a particular
+  // time and ordered with the most recent first.
+  repeated Performance performance = 3;
+}
+
+// The supported operations on the hardware.
+message GateSet {
+  // Prototypes of the gate plus the qubits upon which the gate acts.
+  // If the gate supports parameterized gates, then the parameterized float
+  // will have a unique symbol for this parameter. All operations specify the
+  // qubits they act upon.
+  //
+  // Some of the prototype operation fields will not be set. These include
+  // `incremental_delay_picoseconds`, and `key` and `invert_mask` on measurement
+  // gates.
+  repeated Operation prototypes = 1;
+
+  // For the parameterized gates in prototypes, a map from the parameter name
+  // to the accuracy with which the hardware supports specifying this
+  // parameter. Note that this accuracy may be different than the actual
+  // gate fidelity of this gate, this is just the practical limit that the
+  // hardware can support.
+  ParameterDict accuracies = 2;
+}
+
+// A snapshot of the performance of the hardware.
+message Performance {
+  // The time at which the performance was measured, in micros since epoch.
+  uint64 update_time_micros = 1;
+
+  // A list of the calibrated operations. Note that a given gate from the
+  // hardware's gate set may be calibrated for many different parameter
+  // values.
+  repeated CalibratedOperation calibrated_ops = 2;
+}
+
+// The performance of a calibrated.
+message CalibratedOperation {
+  // Which operation (gate plus placed qubits) was calibrated. If the gate is
+  // not parameterized then this will match exactly the prototype from the
+  // gate set. If the gate was parameterized, then this will be the
+  // actual parameter values (up to accuracy) that the calibration was done for.
+  Operation op = 1;
+
+  // Simple fidelities for the operation.
+  Fidelities fidelities = 2;
+}
+
+// Simple measurements of the performance of a gate.
+message Fidelities {
+  // If the gate is unitary, this is the average gate fidelity.
+  repeated float gate_fidelity = 1;
+
+  // The measurement results for a given preparation.
+  message Confusion {
+    // The key in this map is the measurement result and the value is the
+    // probability.  These may not sum to 1.0 due to leakage.
```

Comment:
I'm not sure that including a caveat here about leakage is particularly useful. Given that the measurements we have defined always return a 0 or 1, there is no _observable_ leakage from the user's perspective, just a probability of measuring state 0 as a 1 or vice-versa. Thus I would suggest replacing `confusion` with a `map <bool, float> measurement_error` that gives the observable measurement error for the 0 and 1 states, respectively.

(Note that for gate fidelities we are also hiding "leakage errors" inside a single gate fidelity number; we could of course expose a richer error model for both gates and measurements in the future.)

Reply:
OK, created a separate MeasruementError message